src/node_api.h

```diff
@@ -71,8 +73,18 @@ typedef struct {
     }                                                                 \
   EXTERN_C_END
 
+#ifdef __wasm32__
+#define NAPI_MODULE(modname, regfunc)                                          \
+  EXTERN_C_START                                                               \
+  NAPI_MODULE_EXPORT napi_value _napi_register(napi_env env,                   \
+                                               napi_value exports) {           \
+    return regfunc(env, exports);                                              \
+  }                                                                            \
+  EXTERN_C_END
```

Comment:
Why is this change necessary? Just to have a predictable function name?

Reply:
right, the napi js impl calls this as a replacement for node's native_module registration api